addons/l10n_tw_edi_ecpay/models/res_company.py

```diff
@@ -0,0 +1,28 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    l10n_tw_edi_ecpay_staging_mode = fields.Boolean(string="Staging mode")
+    l10n_tw_edi_ecpay_staging_url = fields.Char(string="Staging URL", default="https://einvoice-stage.ecpay.com.tw/B2CInvoice/")
+    l10n_tw_edi_ecpay_production_url = fields.Char(string="Production URL", default="https://einvoice.ecpay.com.tw/B2CInvoice/")
```

Comment:
why do you need fields for these? Just use constants in the code, they are not likely to change.